### PR TITLE
Created sum_of_harmonic_series.py

### PR DESCRIPTION
**Pull request by AkshitGulyan**
Here in this code the formula for Harmonic sum is not used, Sum of the series is calculated by creating a list of the elements in the given Harmonic series and adding all the elements of that list !

**A list l1 is created which contains the elements of the Arithmetic Progression to which the given Harmonic Progression is linked and then a list l2 is created which contains the reciprocal of all the elements of list l1 which means list l2 contains all the elements of the given Harmonic Series.**

[Wikipedia link for better understanding of Harmonic Series](https://en.wikipedia.org/wiki/Harmonic_progression_(mathematics))

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/sum_of_harmonic_series.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        l2 is the given Harmonic Series
+        
+        """
+        
+        i+=1
+        l2=[1/x for x in l1]
+    return sum(l2)
+
+
+def main():
```
[the remark this answers is not in the document]
Comment (AkshitGulyan) [reply maths/sum_of_harmonic_series.py:30]: doctest is provided for main function at line no. 23

File: maths/sum_of_harmonic_series.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  print(sum_of_hp(1/2,2,2))
+
+
+if __name__ == "__main__":
+    import doctest
+    doctest.testmod()
```
Comment (AkshitGulyan) [inline maths/sum_of_harmonic_series.py]: here is the doctest

File: maths/sum_of_harmonic_series.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+def sum_of_hp(first_term, common_difference, no_of_terms):
```
Comment (ChrisO345) [inline maths/sum_of_harmonic_series.py:1]:
```suggestion
def sum_of_hp(first_term: float, common_difference: float, no_of_terms: int) -> float:
```

File: maths/sum_of_harmonic_series.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    """
+
+    Find the sum of n terms in an harmonic progression.
+    common_diff is the common difference of Arithmetic
+    Progression by which the given Harmonic Progression is linked.
+
+
```
Comment (ChrisO345) [inline maths/sum_of_harmonic_series.py:9]: add doctests here: https://docs.python.org/3/library/doctest.html

File: maths/sum_of_harmonic_series.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        """
+
+        i += 1
+        l2 = [1 / x for x in l1]
+    return sum(l2)
+
+
+def main():
+    print(sum_of_hp(1 / 2, 2, 2))
```
Comment (ChrisO345) [inline maths/sum_of_harmonic_series.py:31]:
```suggestion

```
this isn't needed

File: maths/sum_of_harmonic_series.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
```
Comment (ChrisO345) [inline maths/sum_of_harmonic_series.py:37]:
```suggestion
    import doctest
    doctest.testmod()
    
    print(sum_of_hp(1 / 2, 2, 2))
```

**Review by ChrisO345 — CHANGES_REQUESTED** [review]

File: maths/sum_of_harmonic_series.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    Progression by which the given Harmonic Progression is linked.
+
+
+    """
+
+    l1 = [1 / first_term]
+    i = 0
+    first_term = 1 / first_term
+    while i < no_of_terms - 1:
+        first_term = first_term + common_difference
```
Comment (cclauss) [inline maths/sum_of_harmonic_series.py:16]:
```suggestion
        first_term += common_difference
```

File: maths/sum_of_harmonic_series.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    """
+
+    Find the sum of n terms in an harmonic progression.
+    common_diff is the common difference of Arithmetic
+    Progression by which the given Harmonic Progression is linked.
+
+
+    """
+
+    l1 = [1 / first_term]
```
Comment (cclauss) [inline maths/sum_of_harmonic_series.py:12]: We require self-documenting variable names.
```suggestion
    arithmetic_progression = [1 / first_term]
```

File: maths/sum_of_harmonic_series.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        l1.append(first_term)
+
+        """
+        l1 is the Arithmetic Progression linked to the given Harmonic Series
+        l2 is the given Harmonic Series
+
+        """
+
+        i += 1
+        l2 = [1 / x for x in l1]
```
Comment (cclauss) [inline maths/sum_of_harmonic_series.py:26]:
```suggestion
        harmonic_series = [1 / step for step in arithmetic_progression]
```

File: maths/sum_of_harmonic_series.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
+def sum_of_harmonic_progression(
```
[the remark this answers is not in the document]
Comment (AkshitGulyan) [reply maths/sum_of_harmonic_series.py:1]: Test File added !

[the remark this answers is not in the document]
Comment (AkshitGulyan) [reply maths/sum_of_harmonic_series.py:1]: Added !

File: maths/sum_of_harmonic_series.py [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
```
[the remark this answers is not in the document]
Comment (AkshitGulyan) [reply maths/sum_of_harmonic_series.py:1]: Added!

[the remark this answers is not in the document]
Comment (AkshitGulyan) [reply maths/sum_of_harmonic_series.py:1]: Added !

[the remark this answers is not in the document]
Comment (AkshitGulyan) [reply maths/sum_of_harmonic_series.py:1]: Added !

**Review by cclauss — APPROVED** [review]
Added two doctests

File: maths/sum_of_harmonic_series.py (hunk 4 above)
Comment (cclauss) [reply maths/sum_of_harmonic_series.py:10]: These are doctests.

**Pull request merged**